enterprise/migrations/0028_link_enterprise_to_enrollment_template.py

```diff
@@ -15,7 +15,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='enrollmentnotificationemailtemplate',
             name='enterprise_customer',
-            field=models.OneToOneField(related_name='enterprise_enrollment_template', default=None, to='enterprise.EnterpriseCustomer'),
+            field=models.OneToOneField(related_name='enterprise_enrollment_template', default=None, to='enterprise.EnterpriseCustomer', on_delete=models.CASCADE),
```

Comment:
Were these old migrations intentionally updated?

Reply:
Yes, Django was throwing deprecation warnings for each migration, creating a lot of noise.

Reply:
Thanks for eliminating the noise. I'm not sure we want to change these to cascading though. It looks like all of our deletes are set to RESTRICT, so I think `models.PROTECT` is what we want although I'm no expert here...
```
mysql> SELECT DELETE_RULE, TABLE_NAME, REFERENCED_TABLE_NAME from information_schema.REFERENTIAL_CONSTRAINTS where TABLE_NAME='enterprise_enrollmentnotificationemailtemplate';
+-------------+------------------------------------------------+-------------------------------+
| DELETE_RULE | TABLE_NAME                                     | REFERENCED_TABLE_NAME         |
+-------------+------------------------------------------------+-------------------------------+
| RESTRICT    | enterprise_enrollmentnotificationemailtemplate | enterprise_enterprisecustomer |
+-------------+------------------------------------------------+-------------------------------+
1 row in set (0.01 sec)
```

I vaguely remember someone saying edX shys away from cascading deletes, maybe for unexpected performance implications but I don't entirely remember the rationale. @feanil or @nasthagiri might know?

Reply:
Is it because we like to have lots of orphaned child records with no clear parent record?  😉

Reply:
I'm not aware of our standpoint on cascading deletes.  In addition to devops' concerns, there may be considerations from the data-analytics perspective as well.

Regardless of that best practice (which I agree we should define and clarify), I caution us from updating past migrations, especially in this way, that we have already run on our servers.  Since we will not run these old migrations on our servers and on pre-existing data, it may cause issues on other open edX instances (or even our own servers if we ever need to migrate back).

The Django-upgrade team may shed further light on this if we really need to address this noise at this time.